.github/workflows/examples-esp32.yaml

```diff
@@ -61,53 +61,69 @@ jobs:
+                     example_binaries/$BUILD_TYPE-build/chip-all-clusters-app.elf
+            - name: Build example All Clusters App C3
+              timeout-minutes: 10
+              run: scripts/examples/esp_example.sh all-clusters-app sdkconfig_c3devkit.defaults
             - name: Copy aside build products
               run: |
                   mkdir -p example_binaries/$BUILD_TYPE-build
                   cp examples/all-clusters-app/esp32/build/chip-all-clusters-app.elf \
                      example_binaries/$BUILD_TYPE-build/chip-all-clusters-app.elf
             - name: Build example Pigweed App
               timeout-minutes: 5
-              run: scripts/examples/esp_example.sh pigweed-app
+              run: scripts/examples/esp_example.sh pigweed-app sdkconfig.defaults
             - name: Copy aside build products
               run: |
                   mkdir -p example_binaries/$BUILD_TYPE-build
                   cp examples/pigweed-app/esp32/build/chip-pigweed-app.elf \
                      example_binaries/$BUILD_TYPE-build/chip-pigweed-app.elf
             - name: Build example Lock App
               timeout-minutes: 5
-              run: scripts/examples/esp_example.sh lock-app
+              run: scripts/examples/esp_example.sh lock-app sdkconfig.defaults
             - name: Copy aside build products
               run: |
                   mkdir -p example_binaries/$BUILD_TYPE-build
                   cp examples/lock-app/esp32/build/chip-lock-app.elf \
                      example_binaries/$BUILD_TYPE-build/chip-lock-app.elf
             - name: Build example Persistent Storage App
               timeout-minutes: 5
-              run: scripts/examples/esp_example.sh persistent-storage
+              run: scripts/examples/esp_example.sh persistent-storage sdkconfig.defaults
             - name: Copy aside build products
               run: |
                   mkdir -p example_binaries/$BUILD_TYPE-build
                   cp examples/persistent-storage/esp32/build/chip-persistent-storage.elf \
                      example_binaries/$BUILD_TYPE-build/chip-persistent-storage.elf
             - name: Build example Shell App
               timeout-minutes: 5
-              run: scripts/examples/esp_example.sh shell
+              run: scripts/examples/esp_example.sh shell sdkconfig.defaults
             - name: Copy aside build products
               run: |
                   mkdir -p example_binaries/$BUILD_TYPE-build
                   cp examples/shell/esp32/build/chip-shell.elf \
                      example_binaries/$BUILD_TYPE-build/chip-shell.elf
             - name: Build example Temperature Measurement App
               timeout-minutes: 25
-              run: scripts/examples/esp_example.sh temperature-measurement-app
+              run: scripts/examples/esp_example.sh temperature-measurement-app sdkconfig.optimize.defaults
```

Comment:
Similarly, here sdkconfig.optimize.defaults has a couple of extra options as compared to sdkconfig.defaults. So, if this compiles fine, then we can safely assume that the temperature-measurement-app with sdkconfig.defaults will compile fine as well